### PR TITLE
done solutions

### PR DESCRIPTION
**Pull request by alexeynefdef**
Problem:
Nested loop in method isMultiplied() witch means O², unfortunately can't resolve this better way.

### CODE REVIEW

File: src/methods/CustomSquare.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package methods;
+
+public class CustomSquare {
+    public int getSquare(int a) {
+        if (a > 1) {
+            a = getSquare(--a) + a + a +1;
```
Comment (sharf85) [inline src/methods/CustomSquare.java:6]: create a new variable for the result

File: src/methods/ProductOfIntegersInList.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.List;
+
+public class ProductOfIntegersInList {
+
+    // O = n²
+    public boolean isMultiplied(List<Integer> digits,int n) {
+        digits.sort(Integer::compareTo);
+        for (int i = 0; i < digits.size(); i++) {
+            for (int j = 0; j < digits.size(); j++) {
+                if (n == digits.get(i)*digits.get(j)) {
```
Comment (sharf85) [inline src/methods/ProductOfIntegersInList.java:12]: i and j should be different

File: src/methods/ProductOfIntegersInList.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package methods;
+
+import java.util.List;
+
+public class ProductOfIntegersInList {
+
+    // O = n²
+    public boolean isMultiplied(List<Integer> digits,int n) {
+        digits.sort(Integer::compareTo);
```
Comment (sharf85) [inline src/methods/ProductOfIntegersInList.java:9]: this is not needed. By the contract the digits are presorted. Besides your solution does not use this fact.

File: src/test/CustomSquareTest.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import methods.CustomSquare;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CustomSquareTest {
+
+    CustomSquare sqr = new CustomSquare();
+
```
Comment (sharf85) [inline src/test/CustomSquareTest.java:11]: add some more tests. You always should check edge cases. For instance, n = 1

File: src/test/ListDuplicatesTest.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ListDuplicatesTest {
+
+    ListDuplicates ld = new ListDuplicates();
+
+    @Test
+    void getDuplicates() {
```
Comment (sharf85) [inline src/test/ListDuplicatesTest.java:17]: make the test to be not sensible to the order. Meaning, your method theoretically can yield ether {2,4} or {4,2}

File: src/test/NODTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import methods.NOD;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class NODTest {
+
+    NOD nod = new NOD();
+
```
Comment (sharf85) [inline src/test/NODTest.java:11]: add more complicated cases

File: src/test/ProductOfIntegersInListTest.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ProductOfIntegersInListTest {
+
+    ProductOfIntegersInList product = new ProductOfIntegersInList();
+    ArrayList<Integer> input = new ArrayList<>(Arrays.asList(1,3,8,9,11,20));
+
```
Comment (sharf85) [inline src/test/ProductOfIntegersInListTest.java:17]: add more tests. Don't forget about edge cases. Test 64
